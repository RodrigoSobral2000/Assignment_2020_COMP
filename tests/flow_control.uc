/* Neste teste analizamos
    os
    *controlos de fluxo* dos programas      aqui
// inseridos
*/



short main      (void) {
    chr letra_abc_1= 'a';

    whil (1) {
        while (letra_abc_1 < 'z') {
            if (letra_abc_1=='j') break ;
            letra_abc_1++;
        }

    return 0;
}
    

void teste(void) {
    int a_b=0;
    short lim3=50;
    
    while ( a_b < lim3 )
        a_b++;

    if (a_b==50) { return 50; }
    else return 0
}