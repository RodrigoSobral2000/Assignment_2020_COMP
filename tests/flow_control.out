SHORT
ID(main)
LPAR
VOID
RPAR
LBRACE
ID(chr)
ID(letra_abc_1)
ASSIGN
CHRLIT('a')
SEMI
ID(whil)
LPAR
INTLIT(1)
RPAR
LBRACE
WHILE
LPAR
ID(letra_abc_1)
LT
CHRLIT('z')
RPAR
LBRACE
IF
LPAR
ID(letra_abc_1)
EQ
CHRLIT('j')
RPAR
RESERVED(break)
SEMI
ID(letra_abc_1)
RESERVED(++)
SEMI
RBRACE
RETURN
INTLIT(0)
SEMI
RBRACE
VOID
ID(teste)
LPAR
VOID
RPAR
LBRACE
INT
ID(a_b)
ASSIGN
INTLIT(0)
SEMI
SHORT
ID(lim3)
ASSIGN
INTLIT(50)
SEMI
WHILE
LPAR
ID(a_b)
LT
ID(lim3)
RPAR
ID(a_b)
RESERVED(++)
SEMI
IF
LPAR
ID(a_b)
EQ
INTLIT(50)
RPAR
LBRACE
RETURN
INTLIT(50)
SEMI
RBRACE
ELSE
RETURN
INTLIT(0)
RBRACE
